### ISSUE TITLE
DO reboots BIGIP system when same configurations/declaration posted

### ISSUE THREAD

**Issue by andreykashcheev** [F5Networks/f5-declarative-onboarding#227]
### Environment
 * Declarative Onboarding Version: 1.21.0
 * BIG-IP Version: 16.0.1.1

### Summary
Please also include information about the reproducibility and the severity/impact of the issue.

On Azure Cloud, while re-processing declaration with identical settings, DO decides to reboot BIGIP system and it causes termination of NodeJS process used for executing Runtime-Init (on-boarding tool) 

### Steps To Reproduce
1. Submit the following declaration (Please note that this was re-run case when the same declaration was posted but with  "ui.advisory.color": "green" whereas in first time, it was "ui.advisory.color": "blue"):
```json
{
  "id": "8f2fbcd6-afaa-42ef-9a2b-50eb8e753b8f",
  "selfLink": "https://localhost/mgmt/shared/declarative-onboarding/task/8f2fbcd6-afaa-42ef-9a2b-50eb8e753b8f",
  "result": {
    "class": "Result",
    "code": 200,
    "status": "OK",
    "message": "success"
  },
  "declaration": {
    "schemaVersion": "1.0.0",
    "class": "Device",
    "label": "Autoscale 1NIC BIG-IP declaration for Declarative Onboarding with PAYG license",
    "async": true,
    "Common": {
      "class": "Tenant",
      "dbVars": {
        "class": "DbVariables",
        "provision.extramb": 1000,
        "restjavad.useextramb": true,
        "ui.advisory.color": "green",
        "ui.advisory.text": "Provisioned via Runtime Init and DO"
      },
      "myDns": {
        "class": "DNS",
        "nameServers": [
          "168.63.129.16"
        ]
      },
      "myNtp": {
        "class": "NTP",
        "servers": [
          "pool.ntp.org"
        ],
        "timezone": "UTC"
      },
      "myProvisioning": {
        "asm": "nominal",
        "class": "Provision",
        "ltm": "nominal"
      },
      "mySystem": {
        "class": "System",
        "autoPhonehome": true,
        "hostname": "aktest-vmss-3.local",
        "consoleInactivityTimeout": 0,
        "cliInactivityTimeout": 0
      }
    }
  }
}

```
2. While re-processing declaration with the same settings, DO decided to reboot BIGIP which causes node process to be terminated. Here is restnoded log:

```
Mon, 03 May 2021 19:58:44 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"kind":"tm:util:bash:runstate","command":"run","utilCmdArgs":"-c \"tmsh modify /sys disk directory /appdata new-size 26128384\"","commandResult":"warning: tm_install::DosPtable::scan_table -- identification of  /dev/sdb1 failed; ID is 7 \n"}
Mon, 03 May 2021 19:58:46 GMT - finest: [telemetry.messageStream:6514.messageStream.tcp] new connection - ::ffff:127.1.1.3-39536
Mon, 03 May 2021 19:58:46 GMT - finest: [telemetry.messageStream:6514.messageStream.tcp] removing connection - ::ffff:127.1.1.1-38194
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"kind":"tm:sys:config:savestate","command":"save","commandResult":"/var/run/f5pcimap: OK\n"}
Mon, 03 May 2021 19:58:48 GMT - info: [f5-declarative-onboarding: restWorker.js] handling REBOOT_NOW event for task: 8f2fbcd6-afaa-42ef-9a2b-50eb8e753b8f
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/echo/available
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/identified-devices/config/device-info/available
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/sys/available
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/cm/available
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/iapp/package-management-tasks/available
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/sys/mcp-state
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"kind":"tm:sys:mcp-state:mcp-statestats","selfLink":"https://localhost/mgmt/tm/sys/mcp-state?ver=16.0.1.1","entries":{"https://localhost/mgmt/tm/sys/mcp-state/0":{"nestedStats":{"entries":{"endPlatformIdReceived":{"description":"true"},"lastLoad":{"description":"full-config-load-succeed"},"phase":{"description":"running"}}}}}}
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/identified-devices/config/device-info
Mon, 03 May 2021 19:58:48 GMT - info: [f5-declarative-onboarding: doUtil.js] Platform: BIG-IP
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"baseMac":"00:0D:3A:59:3A:6E","hostMac":"00:0D:3A:59:3A:6E","halUuid":"5910beaa-d66a-bf41-88fa-96aaa247c395","chassisSerialNumber":"5910beaa-d66a-bf41-96aaa247c395","slots":[{"volume":"HD1.1","product":"BIG-IP","version":"16.0.1.1","build":"0.0.6","isActive":true}],"license":{"registrationKey":"PNUPY-WNGPZ-CNILY-DWVIT-UTPLTEK","activeModules":["MKT, ADD Azure PC|NSHUICJ-MKENAQP","MKT, AWF + LTM, IPI, Threat Campaigns,  25Mbps|WWZCXEE-ZOXWYSQ|Rate Shaping|SDN Services, VE|APM, Limited|SSL, VE|Max Compression, VE|Crypto Offload, VE|MKT, ADD IPI|MKT, ADD Threat Campaigns|Anti-Virus Checks|Base Endpoint Security Checks|Firewall Checks|Network Access|Secure Virtual Keyboard|APM, Web Application|Machine Certificate Checks|Protected Workspace|Remote Desktop|App Tunnel|MKT, AWF + LTM, IPI, Threat Campaigns,  25Mbps"],"generation":0,"lastUpdateMicros":1620065923871037},"interfaces":["1.0","mgmt"],"isIControlRestSupported":true,"icrdPort":8100,"time":1620071928624,"physicalMemory":14336,"platform":"Z100","cpu":"Intel(R) Xeon(R) CPU E5-2673 v3 @ 2.40GHz","machineId":"a16b7413-ea5d-41fd-9f9a-4b35c2bd7f9f","address":"10.0.0.7","hostname":"aktest-vmss-3.local","version":"16.0.1.1","product":"BIG-IP","platformMarketingName":"BIG-IP Virtual Edition","edition":"Point Release 1","build":"0.0.6","restFrameworkVersion":"16.0.1.1-0.0.6","managementAddress":"10.0.0.7","mcpDeviceName":"/Common/aktest-vmss-3.local","isClustered":false,"isVirtual":true,"hypervisorType":"azure","generation":0,"lastUpdateMicros":0,"kind":"shared:resolver:device-groups:deviceinfostate","selfLink":"https://localhost/mgmt/shared/identified-devices/config/device-info"}
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/sys/ready
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"kind":"tm:sys:ready:readystats","selfLink":"https://localhost/mgmt/tm/sys/ready?ver=16.0.1.1","entries":{"https://localhost/mgmt/tm/sys/ready/0":{"nestedStats":{"entries":{"configReady":{"description":"yes"},"licenseReady":{"description":"yes"},"provisionReady":{"description":"yes"}}}}}}
Mon, 03 May 2021 19:58:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] create localhost /tm/sys/config {"command":"save"}
Mon, 03 May 2021 19:58:49 GMT - finest: socket 225 opened
Mon, 03 May 2021 19:58:51 GMT - finest: [telemetry.messageStream:6514.messageStream.tcp] new connection - ::ffff:127.1.1.3-39556
Mon, 03 May 2021 19:58:52 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"kind":"tm:sys:config:savestate","command":"save","commandResult":"/var/run/f5pcimap: OK\n"}
Mon, 03 May 2021 19:58:52 GMT - fine: [f5-declarative-onboarding: doUtil.js] DO state indicates reboot required
Mon, 03 May 2021 19:58:52 GMT - info: [f5-declarative-onboarding: restWorker.js] Reboot required. Rebooting...
Mon, 03 May 2021 19:58:52 GMT - fine: [f5-declarative-onboarding: restWorker.js] create localhost /tm/sys {"command":"reboot"}
Mon, 03 May 2021 19:58:53 GMT - info: [f5-declarative-onboarding: doUtil.js] Platform: BIG-IP
Mon, 03 May 2021 19:58:54 GMT - info: SIGTERM
Mon, 03 May 2021 20:00:01 GMT - info: path /var/config/rest/nodejskeystore/ does not exist
```

I have collected qkview from affected host. Please ping me directly and I will share the link to qkview if needed.

### Expected Behavior

DO does attempts to reboot BIGIP system when same declaration is re-procesed. 

### Actual Behavior
A clear and concise description of what actually happens.
Please include any applicable error output.

After posting the same declaration, DO reboots BIGIP system which terminates running node processes. As a result, on-board process is terminated.

**Comment by dstokesf5** [comment]
Thank you for your feedback. This bug has been added to our internal product backlog as AUTOTOOL-2476.

**Comment by dstokesf5** [comment]
This issue has been resolved by the 1.22.0 release.
